### ISSUE TITLE
No record found for id... when requesting with JWT

### ISSUE THREAD

**Issue by ydeshayes** [feathersjs/feathers#1660]
Hello!

### Steps to reproduce
- Create a new featherjs app with feathersJS CLI, select mongoDB and activate authentication
- POST /users with a new user
- POST /authentication with credential
- GET /users with the token in the Authorization header
- See the error

```
{
    "name": "NotFound",
    "message": "No record found for id '...'",
    "code": 404,
    "className": "not-found",
    "errors": {}
}
```

Code:
https://github.com/ydeshayes/feathers-js-broken

### Expected behavior
It should find the users

### Actual behavior
The authentication service seems to not find the user in mongodb
### System configuration

**Module versions** (especially the part that's not working):
```
 "@feathersjs/authentication": "^4.3.10",
    "@feathersjs/authentication-local": "^4.3.10",
    "@feathersjs/authentication-oauth": "^4.3.10",
    "@feathersjs/configuration": "^4.3.10",
    "@feathersjs/errors": "^4.3.10",
    "@feathersjs/express": "^4.3.10",
    "@feathersjs/feathers": "^4.3.10",
    "@feathersjs/socketio": "^4.3.10",
    "compression": "^1.7.4",
    "cors": "^2.8.5",
    "feathers-mongodb": "^6.0.0",
    "helmet": "^3.21.2",
    "mongodb": "^3.3.3",
    "mongodb-core": "^3.2.7",
    "serve-favicon": "^2.5.0",
    "winston": "^3.2.1"
```

**NodeJS version**:
node version 12.11.1

**Mongodb version**:
4.0.13

**Operating System**:
I tried on 2 differents envs: macos and ubuntu

Thanks!

**Comment by ydeshayes** [comment]
After come digging, it seems like the request done by the _get function in a mongodb service is `db.users.find({ '$and': [ { _id: '5dbbfdc6c8192122806c7f5e' } ] })` instead of `db.users.find({ '$and': [ { _id: ObjectId('5dbbfdc6c8192122806c7f5e') } ] })` whitch come from https://github.com/feathersjs-ecosystem/feathers-mongodb/commit/fe768a29359d203746f206b68c1c8612d2c7fd41

So my question is, how do we convert the id in the authentication service? and why removing that from the mongodb service? That mean I need to convert that and reimplement the get function in my class? Or use a hook?

**Comment by ydeshayes** [comment]
So a hook `ctx.id = new ObjectId(ctx.id);` seems to resolve that. I'll PR the feathers/mongodb doc!

**Comment by daffl** [comment]
So this is probably because of changes in the latest version but what's odd is that all the tests are still passing so this should work with later versions of MongoDB or it needs to be added back to the adapter.

**Comment by ydeshayes** [comment]
this is breaking https://github.com/feathersjs-ecosystem/feathers-authentication-hooks too, we need to had an other hook to objectify the `query.user._id`...

**Comment by ydeshayes** [comment]
So maybe we need to add an option to the adapter to allow devs to disable the objectify if they want only strings, but let the objectify as an opt-out because without that it seems to break a lot of things. 

IMHO it should work out of the box if I just install the adapter with all the libraries.

I can PR the adapter to help

**Comment by magisters-org** [comment]
Facing the same problem. But it covers not only users service, other services can't get records by 'get' method to (on old services, on new generated service i will try now).

**Comment by magisters-org** [comment]
Yeap, all services have this issue with objectifying id on 'get' method. Any thoughts about this?

**Comment by daffl** [comment]
This has been fixed in `feathers-mongodb` v6.1.0

**Comment by mlippert** [comment]
Is this also an issue with `feathers-mongoose`? I'm trying to upgrade from feathers v3 to v4 and mostly I think things work, but tests are broken, I think it's due to `authenticate('jwt') in the hooks.

Actually (and I know this isn't really the right place for a help question) is there a way to easily disable authentication when running from tests?
